runtime/libc/uvwasi_backing.c

```diff
@@ -0,0 +1,1403 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <uvwasi.h>
+
+#include "../runtime.h"
+
+/* Code that actually runs the wasm code */
+IMPORT void wasmf__start(void);
+
+#define unlikely(x) __builtin_expect(!!(x), 0)
+
+static uvwasi_t uvwasi;
+
+/* TODO: This likely makes assumptions about the memory model and needs to be generalized */
+static inline void check_bounds(uint32_t offset, uint32_t bounds_check)
```

Comment:
I think you moved this already but didn't remove it here. Also it needs to go in all memory impls